eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Imp/GovSpec.hs

```diff
@@ -272,7 +275,94 @@ spec =
         expectPulserProposals
         expectNumDormantEpochs 1
         expectExtraDRepExpiry drep 2
+
+    describe "Committee actions validation" $ do
+      it "proposals to update the committee get delayed if the expiration exceeds the max term" $ do
+        setPParams
+        drep <- setupSingleDRep
+        maxTermLength <-
+          getsNES $
+            nesEsL . esLStateL . lsUTxOStateL . utxosGovStateL . curPParamsGovStateL . ppCommitteeMaxTermLengthL
+
+        (initialMembers, govIdCom1) <-
+          impAnn "Initial committee is enacted" $ do
+            c1 <- freshKeyHash
+            c2 <- freshKeyHash
+            currentEpoch <- getsNES nesELL
+            let maxExpiry = currentEpoch + maxTermLength
+            let initialMembers = [(c1, maxExpiry), (c2, maxExpiry)]
+            govIdCom1 <-
+              electCommittee
+                SNothing
+                drep
+                Set.empty
+                initialMembers
+            passEpoch >> passEpoch
+            expectMembers [c1, c2]
+            pure ([c1, c2], govIdCom1)
+
+        (membersExceedingExpiry, exceedingExpiry) <-
+          impAnn "Committee with members exceeding the maxTerm is not enacted" $ do
+            -- submit a proposal for adding two members to the committee,
+            -- one of which has a max term exceeding the maximum
+            c3 <- freshKeyHash
+            c4 <- freshKeyHash
+            currentEpoch <- getsNES nesELL
+            let exceedingExpiry = currentEpoch + maxTermLength + 7
+            let membersExceedingExpiry = [(c3, exceedingExpiry), (c4, currentEpoch + maxTermLength)]
+            _ <-
+              electCommittee
+                (SJust govIdCom1)
+                drep
+                Set.empty
+                membersExceedingExpiry
+            passEpoch >> passEpoch
+            -- the new committee has not been enacted
+            expectMembers initialMembers
+            pure ([c3, c4], exceedingExpiry)
+
+        -- other actions get ratified and enacted
+        govIdConst1 <- impAnn "Other actions are ratified and enacted" $ do
+          (govIdConst1, constitution) <- submitConstitution SNothing
+          submitYesVote_ (DRepVoter (KeyHashObj drep)) govIdConst1
+          hks <- traverse registerCommitteeHotKey (Set.toList initialMembers)
+          traverse_ (\m -> submitYesVote_ (CommitteeVoter (KeyHashObj m)) govIdConst1) hks
+
+          passEpoch >> passEpoch
+          curConstitution <- getsNES $ newEpochStateGovStateL . constitutionGovStateL
+          curConstitution `shouldBe` constitution
+          pure govIdConst1
+
+        -- after enough epochs pass, the expiration of the new members becomes acceptable
+        -- and the new committee is enacted
+        impAnn "New committee is enacted" $ do
+          currentEpoch <- getsNES nesELL
+          let (EpochNo delta) = exceedingExpiry - maxTermLength - currentEpoch
+          replicateM_ (fromIntegral delta) passEpoch
```

Comment:
EpochNo is a Word64, so this calculation can easily overflow resulting in an "infinite" call in `replicateM_`. Not something you want to run into when debugging tests.
```suggestion
          let delta = fromIntegral exceedingExpiry - fromIntegral (maxTermLength + currentEpoch)
          replicateM_ delta passEpoch
```

Reply:
Had to apply `unEpochNo` before `fromIntegral`, otherwise I would get  `Could not deduce (Integral EpochNo)
        arising from a use of ‘fromIntegral’`, which make sense i suppose.